backend/error_status.go

```diff
@@ -0,0 +1,53 @@
+package backend
+
+import (
+	"errors"
+	"net"
+	"net/url"
+	"os"
+	"syscall"
+)
+
+var (
+	connErr *url.Error
+	netErr  *net.OpError
+)
+
+type ErrorDetails struct {
+	Status ErrorStatus
+}
+
+type ErrorStatus int32
+
+const (
+	Unknown ErrorStatus = iota + 1
+	Timeout
+	Unauthorized
+	ConnectionError
```

Comment:
Tricky to come up with all needed. Could https://cloud.google.com/apis/design/errors#handling_errors help us?

Reply:
Why try to outsmart existing HTTP error codes?  seems like just a lot of extra work for less well defined user experience